core/src/main/java/com/tweet/core/model/UserRole.java

```diff
@@ -0,0 +1,19 @@
+package com.tweet.core.model;
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+
+@Entity
+public class UserRole {
+
+    @Id
+    private String name;
```

Comment:
name is actually great candidate for enum. It may be inner enum class in this one.
enum Name {
  USER, ADMINISTRATOR;
}

and this field could be enum instead of string. Also worth considering to change name from name to role (in this case enum should also be Role).

Reply:
and annotate it as you did in User entity for UserRole